static/js/message_flags.js

```diff
@@ -109,8 +109,11 @@ exports.toggle_starred_and_update_server = function (message) {
     }
 };
 
-exports.unstar_all_messages = function () {
-    const starred_msg_ids = starred_messages.get_starred_msg_ids();
+exports.unstar_messages = function (topic_name, stream_id) {
+    let starred_msg_ids = starred_messages.get_starred_msg_ids();
+    if (topic_name && stream_id) {
```

Comment:
I think an if-else statement for setting `starred_msg_ids` will make this function more clear and optimized.